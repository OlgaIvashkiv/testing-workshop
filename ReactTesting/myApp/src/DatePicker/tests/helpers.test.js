import {abbreviationForWeekday, getMonthName, getWeeksForMonth}  from '../helpers';

describe('DatePicker helpers', () => {
  describe('abbreviationForWeekday', () => {
    it('gets the abbreviation', () => {
      const abbreviation = abbreviationForWeekday('Sunday');
      expect(abbreviation).toBe('Su');
    });
  });

  describe('getMonthName', () => {
    it('gets the month name', () => {
      const monthName = getMonthName(4);
      expect(monthName).toBe('May');
    });
  });

  describe('getWeeksForMonth', () => {
    it('gets the weeks', () => {
      const weeks = getWeeksForMonth(1, 2018);
      expect(weeks).toMatchSnapshot();
    });

    it('gets the weeks for leap year', () => {
      const weeks = getWeeksForMonth(1, 2016);
      expect(weeks).toMatchSnapshot();
    });
  });
});
