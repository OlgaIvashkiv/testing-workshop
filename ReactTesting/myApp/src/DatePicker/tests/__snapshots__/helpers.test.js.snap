// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePicker helpers getWeeksForMonth gets the weeks 1`] = `
Array [
  Array [
    null,
    null,
    null,
    null,
    2018-02-01T05:00:00.000Z,
    2018-02-02T05:00:00.000Z,
    2018-02-03T05:00:00.000Z,
  ],
  Array [
    2018-02-04T05:00:00.000Z,
    2018-02-05T05:00:00.000Z,
    2018-02-06T05:00:00.000Z,
    2018-02-07T05:00:00.000Z,
    2018-02-08T05:00:00.000Z,
    2018-02-09T05:00:00.000Z,
    2018-02-10T05:00:00.000Z,
  ],
  Array [
    2018-02-11T05:00:00.000Z,
    2018-02-12T05:00:00.000Z,
    2018-02-13T05:00:00.000Z,
    2018-02-14T05:00:00.000Z,
    2018-02-15T05:00:00.000Z,
    2018-02-16T05:00:00.000Z,
    2018-02-17T05:00:00.000Z,
  ],
  Array [
    2018-02-18T05:00:00.000Z,
    2018-02-19T05:00:00.000Z,
    2018-02-20T05:00:00.000Z,
    2018-02-21T05:00:00.000Z,
    2018-02-22T05:00:00.000Z,
    2018-02-23T05:00:00.000Z,
    2018-02-24T05:00:00.000Z,
  ],
  Array [
    2018-02-25T05:00:00.000Z,
    2018-02-26T05:00:00.000Z,
    2018-02-27T05:00:00.000Z,
    2018-02-28T05:00:00.000Z,
    null,
    null,
    null,
  ],
]
`;

exports[`DatePicker helpers getWeeksForMonth gets the weeks for leap year 1`] = `
Array [
  Array [
    null,
    2016-02-01T05:00:00.000Z,
    2016-02-02T05:00:00.000Z,
    2016-02-03T05:00:00.000Z,
    2016-02-04T05:00:00.000Z,
    2016-02-05T05:00:00.000Z,
    2016-02-06T05:00:00.000Z,
  ],
  Array [
    2016-02-07T05:00:00.000Z,
    2016-02-08T05:00:00.000Z,
    2016-02-09T05:00:00.000Z,
    2016-02-10T05:00:00.000Z,
    2016-02-11T05:00:00.000Z,
    2016-02-12T05:00:00.000Z,
    2016-02-13T05:00:00.000Z,
  ],
  Array [
    2016-02-14T05:00:00.000Z,
    2016-02-15T05:00:00.000Z,
    2016-02-16T05:00:00.000Z,
    2016-02-17T05:00:00.000Z,
    2016-02-18T05:00:00.000Z,
    2016-02-19T05:00:00.000Z,
    2016-02-20T05:00:00.000Z,
  ],
  Array [
    2016-02-21T05:00:00.000Z,
    2016-02-22T05:00:00.000Z,
    2016-02-23T05:00:00.000Z,
    2016-02-24T05:00:00.000Z,
    2016-02-25T05:00:00.000Z,
    2016-02-26T05:00:00.000Z,
    2016-02-27T05:00:00.000Z,
  ],
  Array [
    2016-02-28T05:00:00.000Z,
    2016-02-29T05:00:00.000Z,
    null,
    null,
    null,
    null,
    null,
  ],
]
`;
